component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/voiceotp/VoiceOTPUtils.java

```diff
@@ -0,0 +1,562 @@
+                    "to User Store. Authentication is failed. ", e);
+        }
+    }
+
+    /**
+     * Verify whether user Exist in the user store or not.
+     *
+     * @param username The Username.
+     * @throws VoiceOTPException Exception.
+     */
+    public static void verifyUserExists(String username, String tenantDomain) throws VoiceOTPException,
+            AuthenticationFailedException {
+
+        boolean isUserExist = false;
+        try {
+            UserRealm userRealm = VoiceOTPUtils.getUserRealm(tenantDomain);
+            if (userRealm == null) {
+                throw new VoiceOTPException("Super tenant realm not loaded.");
+            }
+            UserStoreManager userStoreManager = userRealm.getUserStoreManager();
+            if (userStoreManager.isExistingUser(username)) {
+                isUserExist = true;
+            }
+        } catch (UserStoreException e) {
+            throw new VoiceOTPException("Error while validating the user.", e);
+        }
+        if (!isUserExist) {
+
+            log.debug("User does not exist in the User Store");
+            throw new VoiceOTPException("User does not exist in the User Store.");
+        }
+    }
+
+    /**
+     * Get the user realm of the logged in user.
+     *
+     * @param tenantDomain The tenantDomain.
+     * @return The user realm.
+     * @throws AuthenticationFailedException Exception.
+     */
+    public static UserRealm getUserRealm(String tenantDomain) throws AuthenticationFailedException {
+
+        UserRealm userRealm;
+        try {
+            int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+            RealmService realmService = IdentityTenantUtil.getRealmService();
+            userRealm = realmService.getTenantUserRealm(tenantId);
+        } catch (Exception e) {
+            throw new AuthenticationFailedException("Cannot find the user realm for the tenant domain "
+                    + tenantDomain, e);
+        }
+        return userRealm;
+    }
+
+    /**
+     * Get the mobile number for Username.
+     *
+     * @param username The username.
+     * @return Mobile number.
+     * @throws VoiceOTPException Exception.
+     */
+    public static String getMobileNumberForUsername(String username) throws VoiceOTPException,
+            AuthenticationFailedException {
+
+        UserRealm userRealm;
+        String mobile;
+        try {
+            String tenantDomain = MultitenantUtils.getTenantDomain(username);
+            String tenantAwareUsername = MultitenantUtils.getTenantAwareUsername(username);
+            userRealm = getUserRealm(tenantDomain);
+            if (userRealm == null) {
+                throw new VoiceOTPException("Cannot find the user realm for the given tenant domain : " + tenantDomain);
+            } else {
+                mobile = userRealm.getUserStoreManager()
+                        .getUserClaimValue(tenantAwareUsername, VoiceOTPConstants.MOBILE_CLAIM, null);
+            }
```

Comment:
No need to have an else block. Please put the rest of the code after the if block.